src/utils/providerFetch.ts

```diff
@@ -0,0 +1,129 @@
+/**
```

Comment:
Move this file into src/retriever, and don't use camelCase for filenames, maybe just make it utils.ts in there